manager.c

```diff
@@ -160,11 +160,35 @@ void update_jobs(void)
 	job_manifest_t jm;
 	job_t job, job_tmp;
 	LIST_HEAD(,job) joblist;
+	bool skip;
 
 	LIST_INIT(&joblist);
 
 	/* Pass #1: load all jobs */
 	LIST_FOREACH(jm, &pending, jm_le) {
+		skip = false;
+		LIST_FOREACH(job, &jobs, joblist_entry) {
+			if (!strcmp(jm->label, job->jm->label)) {
+				log_warning("job `%s' is already running", jm->label);
+				skip = true;
+				break;
+			}
+		}
+
+		if (skip)
+			continue;
+		
+		LIST_FOREACH(job, &jobs, joblist_entry) {
```

Comment:
From here down, it looks like duplicate code with a slightly different warning message.